containers/src/Data/IntSet/Internal/IntTreeCommons.hs

```diff
@@ -161,6 +162,14 @@ i2w :: Int -> Word
 i2w = fromIntegral
 {-# INLINE i2w #-}
 
+-- Used to compare IntSets and IntMaps
+data Order
+  = Less       -- holds for [0,3,4] [0,3,5,1]
+  | Prefix'    -- holds for [0,3,4] [0,3,4,5]
+  | Equals     -- holds for [0,3,4] [0,3,4]
+  | FlipPrefix -- holds for [0,3,4] [0,3]
+  | Greater    -- holds for [0,3,4] [0,2,5]
+
```

Comment:
The naming is ... not trivial. What do you think about something like this?

```haskell
data Order
  = A_LT_B
  | A_Prefix_B
  | A_EQ_B
  | B_Prefix_A
  | A_GT_B
```

Reply:
Seems fine, changed.